packages/synapse-react-client/src/components/Authentication/LoginForm.tsx

```diff
@@ -32,6 +33,8 @@ type Props = {
   ssoState?: OAuth2State
   /* The URI where the user should be directed in an email when attempting to reset 2FA */
   twoFactorAuthResetUri?: string
+  /* Invoked when password login is selected */
+  onPasswordLoginSelected?: () => void
```

Comment:
Add a callback so we can determine when to render our special password + disable 2FA form while still using the shared `LoginForm` & `AuthenticationMethodSelection` components